src/core/ddsc/src/dds_qos.c

```diff
@@ -335,6 +335,151 @@ void dds_qset_ignorelocal (dds_qos_t * __restrict qos, dds_ignorelocal_kind_t ig
   qos->present |= QP_CYCLONE_IGNORELOCAL;
 }
 
+static void dds_qprop_init (dds_qos_t * qos)
+{
+  if (!(qos->present & QP_PROPERTY_LIST))
+  {
+    qos->property.value.n = 0;
+    qos->property.value.props = NULL;
+    qos->property.binary_value.n = 0;
+    qos->property.binary_value.props = NULL;
+    qos->present |= QP_PROPERTY_LIST;
+  }
+}
+
+static bool dds_qprop_get_index (const dds_qos_t * qos, const char * name, uint32_t * index)
+{
+  bool found = false;
+  if (qos == NULL || name == NULL || index == NULL || !(qos->present & QP_PROPERTY_LIST))
+    return false;
+  for (uint32_t i = 0; !found && i < qos->property.value.n; i++)
+  {
+    if (strcmp (qos->property.value.props[i].name, name) == 0)
+    {
+      *index = i;
+      found = true;
+    }
+  }
+  return found;
+}
+
+static bool dds_qbprop_get_index (const dds_qos_t * qos, const char * name, uint32_t * index)
```

Comment:
I would merge the bulk of the two ``get_index`` functions: they are identical save for ``value.X`` vs ``binary_value.X``. That can be parameterised.

Reply:
Parameterisation to a single function is hard because of the two distinct types `dds_property_t` and `dds_binaryproperty_t`, this would required struct-length and offset of the name field as parameters to that function, which makes the code harder to understand. Because I also really didn't like the code duplication, I've created macros for get_index, unset and get_propnames